### ISSUE TITLE
Hope that Message component could support custom icon

### ISSUE THREAD

**Issue by ishowman** [NG-ZORRO/ng-zorro-antd#3123]
## What problem does this feature solve?
There're some case that the default icons of Message component is not suitable for our products, it's necessary to add an api for developer to use custom icons.
## What does the proposed API look like?
I hope that add an new api for Message component like below:
```
this.message.success('Loading finished', 
  { nzCustomIcon: 'icon template ref or html like <i class="fas fa-angle-left"></i>' }
)
```
Maybe there are better solutions, hoping to get reply from you. Thanks!

**Comment by wendzhue** [comment]
@ishowman You can track this PR. You would be able to put an icon in the template and use `messageService.create('', templateWithIcon, options)` to achieve that.

**Comment by wendzhue** [comment]
Closed by #3102.
